diesel_cli/src/main.rs

```diff
@@ -606,6 +607,31 @@ fn regenerate_schema_if_file_specified(
     Ok(())
 }
 
+// List which features have been compiled into this version of diesel_cli
+#[cfg(any(feature = "postgres", feature = "mysql", feature = "sqlite"))]
+fn show_which_engines_supported() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+    let mut features: Vec<&str> = Vec::new();
+    #[cfg(feature = "postgres")]
+    features.push("postgres");
+    #[cfg(feature = "mysql")]
+    features.push("mysql");
+    #[cfg(feature = "sqlite")]
+    features.push("sqlite");
+
+    for f in &features {
+        match f as &str {
+            "postgres" | "sqlite" | "mysql" | "sqlite-bundled" => print!("{} ", f),
+            other => print!("'{}' ", other),
+        };
+    }
+    if features.is_empty() {
+        println!("[warning] No features detected");
+    } else {
+        println!();
```

Comment:
Why do we need this `match` statement? We can just do `println!("Features: {}", features.join(", "))`

Reply:
None of the features have spaces, so the output can be parsed as space delimited by other programs that depend on the output. By adding "Features: " that just increases the complexity of processing the output. I can see that adding "Features: " provides clarity, but in practical use the command run is `diesel features` which should lead the user to expect that the output would be features.

Also by having this part rather than the nifty .join(", ") means that adding a standard `-n     do not output the trailing newline` in the future becomes trivial.

Reply:
Okay, but we can still do `println!("{}", features.join(" "))`. If you want newline you can manipulate this string inside `join`.

Reply:
Please remove the match and do this and it will be good for merging.